### DISCUSSION TITLE
Working with Hashable type for variable names and mypy

### DISCUSSION THREAD

**Discussion by jorgsk** [pydata/xarray#4834] [General] [answered]
Hi We'd like to run mypy as part of CI of an application that uses xarray but ran into a large number of barks from mypy because we perform string-operations on variable names from Datasets and DataArrays. These are type-defined as Hashable.

What would be the best way to work around something like:

```
with xr.open_dataset('my_ncfile.nc') as ncfile:
    new_names = [var_name.upper() for var_name in ncfile.data_vars]
```
Without adding `# type: ignore` to these lines?

**Comment by rhkleijn** [comment, marked as answer]
I guess `str(var_name).upper()` would do the trick.

**Reply by max-sixty** [reply]
Yes, think unavoidable when the values are read externally.

Potentially there could be some way of using generics to allow retaining type information between the ds construction and later code; others will know more as to whether that's possible

**Reply by jorgsk** [reply]
Thanks! In other places I've used `cast('str', var_name)` where `cast` is from the Typing module. I ended up sprinkling these changes in a handful of places and now mypy is happy.
